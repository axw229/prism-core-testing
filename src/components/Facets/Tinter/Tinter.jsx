// @flow
import React from 'react'

import LivePalette from '../../LivePalette/LivePalette'
import SceneManager from '../../SceneManager/SceneManager'

import { withDragDropContext } from '../../../helpers/WithDragDropContext'

type Props = {
  sceneSet?: string
}

function Tinter ({ sceneSet }: Props) {
  return (
    <React.Fragment>
      <SceneManager type={sceneSet} />
      <LivePalette />
    </React.Fragment>
  )
}

export default withDragDropContext(Tinter)
