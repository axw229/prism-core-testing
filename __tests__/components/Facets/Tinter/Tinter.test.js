/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Tinter } from 'src/components/Facets/Tinter/Tinter'

const sceneSet = 'rooms'
const getTinter = (props) => {
  let defaultProps = {
    sceneSet: sceneSet
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<Tinter {...newProps} />)
}

describe('Tinter without props', () => {
  let tinter
  beforeAll(() => {
    if (!tinter) {
      tinter = getTinter({ sceneSet: '' })
    }
  })

  it('should match snapshot', () => {
    expect(tinter).toMatchSnapshot()
  })

  it('should render', () => {
    expect(tinter.exists()).toBe(true)
  })

  it('should render SceneManager with Connect', () => {
    expect(tinter.find('Connect(SceneManager)').exists()).toBe(true)
  })

  it('should render LivePalette with Connect', () => {
    expect(tinter.find('Connect(LivePalette)').exists()).toBe(true)
  })
})

describe('Tinter with props', () => {
  let tinter
  beforeAll(() => {
    if (!tinter) {
      tinter = getTinter()
    }
  })

  it('should match snapshot', () => {
    expect(tinter).toMatchSnapshot()
  })

  it('should render', () => {
    expect(tinter.exists()).toBe(true)
  })

  it('should render SceneManager with Connect', () => {
    expect(tinter.find('Connect(SceneManager)').exists()).toBe(true)
  })

  it('should render LivePalette with Connect', () => {
    expect(tinter.find('Connect(LivePalette)').exists()).toBe(true)
  })

  it('should render SceneManager with Connect & prop type defined as sceneSet constant', () => {
    expect(tinter.find('Connect(SceneManager)').prop('type')).toEqual(sceneSet)
  })
})
