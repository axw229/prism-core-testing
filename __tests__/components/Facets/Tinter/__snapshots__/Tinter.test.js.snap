// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tinter with props should match snapshot 1`] = `
<Fragment>
  <Connect(SceneManager)
    type="rooms"
  />
  <Connect(LivePalette) />
</Fragment>
`;

exports[`Tinter without props should match snapshot 1`] = `
<Fragment>
  <Connect(SceneManager)
    type=""
  />
  <Connect(LivePalette) />
</Fragment>
`;
